tfx/components/schema_gen/executor.py

```diff
@@ -59,21 +61,37 @@ def Do(self, input_dict: Dict[Text, List[types.Artifact]],
         - output: A list of 'Schema' artifact of size one.
       exec_properties: A dict of execution properties, includes:
         - infer_feature_shape: Whether or not to infer the shape of the feature.
+        - exclude_splits: Names of splits that will not be taken into
+          consideration when auto-generating a schema.
 
     Returns:
       None
     """
     # TODO(zhitaoli): Move constants between this file and component.py to a
     # constants.py.
-    train_stats_uri = io_utils.get_only_uri_in_dir(
-        artifact_utils.get_split_uri(input_dict[STATISTICS_KEY], 'train'))
+    infer_feature_shape = exec_properties['infer_feature_shape']
+    exclude_splits = exec_properties[EXCLUDE_SPLITS_KEY]
+    # Only one schema is generated for all splits.
+    schema = None
+    for artifact in input_dict[STATISTICS_KEY]:
+      for split in artifact_utils.decode_split_names(artifact.split_names):
+        if exclude_splits and split in exclude_splits:
+          continue
+        stats_uri = io_utils.get_only_uri_in_dir(
+            artifact_utils.get_split_uri(input_dict[STATISTICS_KEY], split))
+        if not schema:
+          absl.logging.info('Infering schema from statistics for split '
+                            '{}.'.format(split))
+          schema = tfdv.infer_schema(
+              tfdv.load_statistics(stats_uri), infer_feature_shape)
+        else:
+          absl.logging.info('Updating schema from statistics for split '
```

Comment:
maybe just merge the logging to 'process schema for split xxx'

as we don't care which one use infer which one use update, the order doesn't matter, we just need to know all required splits are processed